### PR TITLE
Add Various Grab Visualization Options

### PR DESCRIPTION
**Pull request by imgntn**
This PR adds a variety of ways to visualize searching and moving distant objects to handControllerGrab.js -- the default will use particles for searching and moving.

var USE_ENTITY_LINES_FOR_SEARCHING = false;
var USE_OVERLAY_LINES_FOR_SEARCHING = false;
var USE_PARTICLE_BEAM_FOR_SEARCHING = true;

var USE_ENTITY_LINES_FOR_MOVING = false;
var USE_OVERLAY_LINES_FOR_MOVING = false;
var USE_PARTICLE_BEAM_FOR_MOVING = true;

var USE_SPOTLIGHT = false;
var USE_POINTLIGHT = false

Overlays can only be seen by you.  Entities, particle systems, and lights can be seen by everyone.
I'll delete additional files once we settle on a combination of features.

To test: 
https://rawgit.com/imgntn/hifi/particle_grab/examples/controllers/handControllerGrab.js

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4e9d81f18737203e4bbdd60343aab98d65a210a1 available at: https://gist.github.com/101c8958c0b6ad9d9813<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/High+Fidelity-PR-6630-win64.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6630.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 11aaeb1ec00a76273f639648d47a00e4f691e0aa available at: https://gist.github.com/101c8958c0b6ad9d9813<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/High+Fidelity-PR-6630-win64.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6630.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1b1edf6f7ab8a8e94e7d78df9eaf2b42d81c703b available at: https://gist.github.com/bb2dd6ee6295e025739c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/High+Fidelity-PR-6630-win64.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6630.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 70a0e51cc9b5898dc49b1134f84a6f96d50b7aff available at: https://gist.github.com/101c8958c0b6ad9d9813<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/High+Fidelity-PR-6630-win64.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6630.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: fe05542440f2c74668cbec2859c2c8227eb91e42 available at: https://gist.github.com/101c8958c0b6ad9d9813<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/High+Fidelity-PR-6630-win64.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6630.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0ddc2ba76e22359f202cb2205efbf6da24ec08a1 available at: https://gist.github.com/101c8958c0b6ad9d9813<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/High+Fidelity-PR-6630-win64.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6630.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: a0b698c0283ce162720ad13465d04ccb8105439d available at: https://gist.github.com/101c8958c0b6ad9d9813<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/High+Fidelity-PR-6630-win64.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6630.dmg

**Comment by ericrius1** [comment]
The particle beams are awesome. I was able to get a smoother looking line by changing these two props:
"particleRadius": 0.015,
 "radiusSpread": 0.005,

and also setting spread on line 457 to 2 instead of 0

**Comment by ericrius1** [comment]
I intuitively like the point light better than the spotlight. With the latter, I feel like I'm being told were to drop the object I'm holding, as opposed to the point light, which just generally calls out the object in question.

**Comment by ericrius1** [comment]
Overall- I really like the particles and the point light. My ideal scenario would be particles all the time, but unless an object is grabbed, the particles are local only, and once object is grabbed, they appear for remote clients. Since we don't have particle overlays, I'd just have particles all the time, and keep the point light behavior as it is in https://rawgit.com/imgntn/hifi/particle_grab/examples/controllers/handControllerGrab-pointlight.js

File: examples/controllers/handControllerGrab.js [hunk 1]
```diff
@@ -409,6 +401,239 @@ function MyController(hand) {
...
+            "alphaFinish": 1,
+            "additiveBlending": 0,
+            "textures": "https://hifi-content.s3.amazonaws.com/alan/dev/textures/grabsprite-3.png"
+        }
+
+        this.particleBeam = Entities.addEntity(particleBeamProperties);
+    };
+
+    this.updateParticleBeam = function(position, orientation, color, speed, spread, lifespan) {
+        print('lifespan::' + lifespan);
```
Comment (sethalves) [inline examples/controllers/handControllerGrab.js]: left-over debugging print

**Comment by hifi-gustavo** [comment]
Build console output for commit: 64eb22bf228639d7ff79ec0917eef70cf4f92fe8 available at: https://gist.github.com/101c8958c0b6ad9d9813<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6630.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6630.dmg

**Pull request merged**